pkg/dogstatsd/server.go

```diff
@@ -320,6 +320,7 @@ func (s *Server) parsePackets(batcher *batcher, packets []*listeners.Packet) {
 				}
 			}
 		}
+		s.packetPool.Put(packet)
```

Comment:
The way the `packetPool` is embedded and dealt with each listener implementation, this was unfortunately an easy mistake to make (I even had trouble ensuring what this PR is fixing).
In the future, we'll probably have to re-visit a bit the `Server` struct and how it interacts with the `Listener` implementation.
PR LGTM though.

Reply:
I agree that this is very easy to miss and could use some refacto